### DISCUSSION TITLE
Segfault with Proof Generation

### DISCUSSION THREAD

**Discussion by fslivovsky** [cvc5/cvc5#10537] [Q&A] [answered]
Hi,

I wanted to try proof generation in the Alethe format, using the following example:

```smt
(set-logic LIA)
(assert (not (forall ((x Int)) (exists ((y Int)) (= x y)))))
(check-sat)
``` 
However, the call

`./bin/cvc5 --dump-proofs --proof-format-mode=alethe --simplification=none --dag-thresh=0 --proof-granularity=theory-rewrite test.smt2`

fails with the following output:

```shell
unsat
cvc5 suffered a segfault.
Offending address is 0x5
Looks like a NULL pointer was dereferenced.
Segmentation fault (core dumped)
``` 

The example used in the docs (`../test/regress/cli/regress0/proofs/qgu-fuzz-1-bool-sat.smt2`) works fine.

Does anyone here have an idea why proof generation fails in the example above?

Thanks,
Friedrich

**Comment by ajreynol** [comment, marked as answer]
Sorry for the confusion here.

cvc5 generates AletheLF proofs by default `(--proof-format=alf)`, which is fully supported on main.

The option `--proof-format=alethe` make cvc5 output proofs in a similar format but with a different proof calculus.  This option is experimental on main.  We are planning to have a robust implementation of this option in the future, which also will use an updated syntax.

In the meantime, I've updated the documentation on main and guarded usage of `--proof-format=alethe` by also requiring the option `--proof-alethe-experimental` (see https://github.com/cvc5/cvc5/pull/10539).

**Reply by fslivovsky** [reply]
Thank you!

**Comment by HanielB** [comment]
@fslivovsky FYI in [this branch](https://github.com/cvc5/cvc5/tree/proof-new) the cvc5 support for Alethe is robust and in particular it produces a valid proof for the problem you posted. Valid here means that it is a proof that is checked by [Carcara](https://github.com/ufmg-smite/carcara), the proof checker for Alethe. Note however that cvc5 produces Alethe proofs that slightly deviate from the [Alethe standard](https://verit.gitlabpages.uliege.be/alethe/specification.pdf), so checking cvc5 proofs with Carcara requires these options for Carcara: `-i --allow-int-real-subtyping --expand-let-bindings`.

Note also that `--simplification=none --dag-thresh=0 --proof-granularity=theory-rewrite` are no longer needed when requesting Alethe proofs from cvc5. 

In the following weeks we expect to merge into main the bug fixes that are currently only in the referenced branch.

**Reply by fslivovsky** [reply]
Thanks!
